### ISSUE TITLE
Attribute cache busting

### ISSUE THREAD

**Issue by ghost** [flyerhzm/simple_cacheable#4]
When a find_cached_by query is run, it looks as if you're storing the value in an array. For example, if I call User.find_cached_by_email('jd@domain.com') , the email address 'jd@domain.com' will be stored in a class attribute.

```
attributes.each do |attribute|
    class_eval <<-EOF
         def self.find_cached_by_#{attribute}(value)
             self.cached_indices["#{attribute}"] ||= []
             self.cached_indices["#{attribute}"] << value  ###  <== HERE
             Rails.cache.fetch attribute_cache_key("#{attribute}", value) do
                 self.find_by_#{attribute}(value)
            end
         end
    EOF
end
```

Later, when a user record is persisted to the database, you're busting the attribute cache keys like this :

```
self.class.cached_indices.each do |attribute, values|
    values.each do |value|
        Rails.cache.delete self.class.attribute_cache_key(attribute, value)
    end
end
```

I'm looking at your code quickly, but it seems that you are going to loop through every value that was ever queried, for each attribute, and bust those cache keys... is that true?

If so, should we be slightly concerned with this? For example, the primary query in our application finds users via User.find_cached_by_email, and we have thousands of unique user records. Each time a single user record is persisted to the db, this code is going to loop an array containing EVERY email address that was ever queried using find_cached_by_email (thousands), and attempt to bust cached keys for those email addresses... 

I think we really just want to attempt to bust a single key per cached_indice, do we not?

```
self.class.cached_indices.each do |attribute|
   value = self.send(attribute)
   Rails.cache.delete self.class.attribute_cache_key(attribute, value)
end
```

We may have to bust 2 keys if the record attribute value has just changed...

```
self.class.cached_indices.each do |attribute|
   if self.send("#{attribute}_changed?")
      value_was = self.send("#{attribute}_was")
      Rails.cache.delete self.class.attribute_cache_key(attribute, value_was)
  end
   value = self.send(attribute)
   Rails.cache.delete self.class.attribute_cache_key(attribute, value)
end
```

I'd be glad to hear your thoughts. I've just finished writing a gem for our internal gem that does almost exactly like you've done, but I prefer your code LOL.

**Comment by flyerhzm** [comment]
oh, you are right, I will fix it soon.

Blame me!

**Comment by flyerhzm** [comment]
it's fixed and I released 1.1.5 version, thanks

**Comment by ghost** [comment]
Great. Thanks for the quick fix. By the way, fantastic clean coding. Love it.
